src/main/webapp/app/modules/canvas/visualization/util/visualization-utils.ts

```diff
@@ -0,0 +1,166 @@
+const getFormattedNumber = (value, format) => {
+  switch (format) {
+    case 'Actual':
+      return value;
+    case 'K':
+      return value / 1000;
+    case 'M':
+      return value / 1000000;
+    case 'B':
+      return value / 1000000000;
+    case 'Percent':
+      return value;
+    default:
+      return value;
+  }
+};
+
+const isFunction = obj => {
+  return !!(obj && obj.constructor && obj.call && obj.apply);
+};
+
+const getDimensionsAndMeasures = fields => {
+  const dimensions = fields
+    .filter(function (item) {
+      return item.feature && item.feature.featureType === 'DIMENSION';
+    })
+    .map(function (item) {
+      const newItem = {};
+      // angular.copy(item, newItem);
+      //newItem.feature.name = newItem.feature.selectedName;
+      return newItem;
+    })
+    .sort(function (a, b) {
+      return sortBySequenceNumber(a.fieldType, b.fieldType);
+    });
+
+  const measures = fields
+    .filter(function (item) {
+      return item.feature && item.feature.featureType === 'MEASURE';
+    })
+    .map(function (item) {
+      const newItem = {};
+      //  angular.copy(item, newItem);
+      //  newItem.feature.name = newItem.feature.name;
+      return newItem;
+    })
+    .sort(function (a, b) {
+      return sortBySequenceNumber(a.fieldType, b.fieldType);
+    });
+  return {
+    measures: measures,
+    dimensions: dimensions,
+  };
+};
+
+const getDimensionsAndMeasuresForNotification = fields => {
+  const dimensions = fields
+    .filter(function (item) {
+      return item.feature && item.feature.featureType === 'DIMENSION';
+    })
+    .map(function (item) {
+      const newItem = {};
+      //   angular.copy(item, newItem);
+      //.feature.name = newItem.feature.name;
+      return newItem;
+    })
+    .sort(function (a, b) {
+      return sortBySequenceNumber(a.fieldType, b.fieldType);
+    });
+
+  const measures = fields
+    .filter(function (item) {
+      return item.feature && item.feature.featureType === 'MEASURE';
+    })
+    .map(function (item) {
+      const newItem = {};
+      //  angular.copy(item, newItem);
+      // newItem.feature.name = newItem.feature.name;
+      return newItem;
+    })
+    .sort(function (a, b) {
+      return sortBySequenceNumber(a.fieldType, b.fieldType);
+    });
+  return {
+    measures: measures,
+    dimensions: dimensions,
+  };
+};
+
+const getPropertyValue = (properties, propertyName, orElse) => {
+  const _properties = properties.filter(function (item) {
```

Comment:
since we are not use _ in every var declaration..avoid using _ in every var declaration

Reply:
done